dynamics/orbital_dynamics.py

```diff
@@ -189,6 +212,28 @@ def perturbed_state_derivative(self, x: np.ndarray, epoch: Epoch = None) -> np.n
 
             updated_a += a_J2_gt
 
+        # Compute J3 and J4
+        if self.use_j34 and not np.isclose(r_norm, 0):
+            a_J3_gt = j3_dynamics(r)
+            a_J4_gt = j4_dynamics(r)
+            print(f"J3: {a_J3_gt}, J4: {a_J4_gt}")
+            updated_a += a_J3_gt + a_J4_gt
+
+        # Compute third body gravity
+        if self.use_sun_grav and not np.isclose(r_norm, 0):
```

Comment:
I don't think you need this extra check since there shouldn't be any division by zero issues here.
```suggestion
        if self.use_sun_grav:
```